napari/_qt/qt_viewer.py

```diff
@@ -808,6 +805,10 @@ def _process_mouse_event(self, mouse_callbacks, event):
             self.view, self.viewer.dims.point, self.viewer.dims.displayed
         )
 
+        # Update the cursor position
+        self.viewer.cursor._view_direction = event.view_direction
+        self.viewer.cursor.position = self._map_canvas2world(list(event.pos))
```

Comment:
Yeah it looks like there is some redundancy here. We moved the settings of `self.viewer.cursor.position` down because the `get_status` is triggered on the `Cursor.position` setter event. Thus, the `Cursor._view_direction` needs to be set first. In any case, I can have a look this afternoon.

Reply:
You were right that we were setting the event with the hold `position`. We fixed the order and added a test that fails when event.position is not set correctly.